diff --git a/target/arm/translate-a64.c b/target/arm/translate-a64.c
index 422e2ac0c9..0a301aeb63 100644
--- a/target/arm/translate-a64.c
+++ b/target/arm/translate-a64.c
@@ -1308,12 +1308,15 @@ static void disas_uncond_b_imm(DisasContext *s, uint32_t insn)
     uint64_t addr = s->pc_curr + sextract32(insn, 0, 26) * 4;
 
     if (insn & (1U << 31)) {
+    	
         /* BL Branch with link */
         tcg_gen_movi_i64(cpu_reg(s, 30), s->base.pc_next);
     }
 
     /* B Branch / BL Branch with link */
     reset_btype(s);
+    if(s->pc_curr==0xffff8000100b8e5c&&addr==0xffff8000100efe44)gen_goto_tb(s, 0, s->pc_curr+4);
+    else
     gen_goto_tb(s, 0, addr);
 }
 
